app/modules/notifications/models.py

```diff
@@ -30,24 +55,118 @@ class Notification(db.Model, HoustonModel):
     guid = db.Column(
         db.GUID, default=uuid.uuid4, primary_key=True
     )  # pylint: disable=invalid-name
-    timestamp = db.Column(db.DateTime, server_default=func.now(), nullable=False)
+
     status = db.Column(
         db.String(length=255), default=NotificationStatus.unread, nullable=False
     )
-    message_template = db.Column(
-        db.String, default=NotificationMessageTemplate.raw, nullable=False
-    )
+    message_type = db.Column(db.String, default=NotificationType.raw, nullable=False)
     message_values = db.Column(db.JSON, nullable=True)
     recipient_guid = db.Column(
         db.GUID, db.ForeignKey('user.guid'), index=True, nullable=True
     )
-    recipient = db.relationship('User')
+    recipient = db.relationship('User', back_populates='notifications')
 
     def __repr__(self):
         return (
             '<{class_name}('
             'guid={self.guid}, '
-            'template={self.message_template}, '
+            'template={self.message_type}, '
             "recipient='{self.recipient}'"
             ')>'.format(class_name=self.__class__.__name__, self=self)
         )
+
+    # returns dictionary of channel:bool
+    def channels_to_send(self):
+        # pylint: disable=invalid-name
+        channels = None
+        user_prefs = UserNotificationPreferences.get_user_preferences(self.recipient)
+        if self.message_type in user_prefs.keys():
+            channels = user_prefs[self.message_type]
+
+        return channels
+
+    @classmethod
+    def create(cls, notification_type, user, data):
```

Comment:
wondering if `data` can have _values_ that are _objects_ instead -- just to kinda save a step/lookup.  e.g. i saw above the user **guid** was getting passed in.   just thinking that in _most_ cases, the template-expanding is almost always going to be grabbing the object.  (User, Encounter, Individual, etc).   but maybe there is a reason (such as storing as json or something); so ignore this if there are Reasons.

Reply:
I was thinking that the call to this method was the place that would have to know how to build the message contents. I was hoping to keep this dumb so that it only checked that the required fields were present as defined by the table. 
This is the decoupling I mean, this code should be about how the Notification object is stored and minimally validated. It should not be building the content... to my mind.

Reply:
that makes sense to me.